internal/cache/cr.go

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Popeye
+
+package cache
+
+import (
+	"sync"
+
+	"github.com/derailed/popeye/internal"
+	"github.com/derailed/popeye/internal/db"
+	rbacv1 "k8s.io/api/rbac/v1"
+)
+
+// ClusterRole represents ClusterRole cache.
+type ClusterRole struct {
+	db *db.DB
+}
+
+// NewClusterRole returns a new ClusterRole cache.
+func NewClusterRole(db *db.DB) *ClusterRole {
+	return &ClusterRole{db: db}
+}
+
+// RoleRefs computes all role external references.
+func (r *ClusterRole) AggregationMatchers(refs *sync.Map) {
+	txn, it := r.db.MustITFor(internal.Glossary[internal.CR])
+	defer txn.Abort()
+	for o := it.Next(); o != nil; o = it.Next() {
+		cr := o.(*rbacv1.ClusterRole)
+		if cr.AggregationRule != nil {
+			for _, lbs := range cr.AggregationRule.ClusterRoleSelectors {
+				for k, v := range lbs.MatchLabels {
```

Comment:
This assumes matchLabels are in effect,  but what if the user leverages matchExpressions instead?

Reply:
matchExpressions are not currently implemented, they are more complex:

```
FIELDS:
   key	<string> -required-
     key is the label key that the selector applies to.

   operator	<string> -required-
     operator represents a key's relationship to a set of values. Valid
     operators are In, NotIn, Exists and DoesNotExist.

   values	<[]string>
     values is an array of string values. If the operator is In or NotIn, the
     values array must be non-empty. If the operator is Exists or DoesNotExist,
     the values array must be empty. This array is replaced during a strategic
     merge patch.
```

I wasn't sure you were interested in the functionality and I don't have it in my setup so I just ignored them. We can either merge it without and add it later or wait until I'm able to add it, What do you prefer?

Reply:
For completeness I think we should support matchExpressions.